core/src/main/java/org/apache/iceberg/actions/FileRewritePlanner.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.ContentFile;
+import org.apache.iceberg.ContentScanTask;
+
+/**
+ * A class for planning content file rewrites.
+ *
+ * <p>The entire rewrite operation is broken down into pieces based on partitioning, and size-based
```

Comment:
"partitioning and size-based" are implementation details right? Not necessarily always true?

Reply:
Copied from [FileRewriter](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/actions/FileRewriter.java#L30), but agree with your assessment, so rewritten the javadoc.

The partitioning is currently necessary (see `FileRewritePlan#groupsInPartition`). Removed the size-based part.